vendor/k8s.io/kubernetes/openshift-kube-apiserver/authentication/oauth/tokenauthenticator.go

```diff
@@ -55,6 +58,18 @@ func (a *tokenAuthenticator) AuthenticateToken(ctx context.Context, name string)
 		groupNames = append(groupNames, group.Name)
 	}
 
+	tokenAudiences := a.implicitAuds
+	requestedAudiences, ok := kauthenticator.AudiencesFrom(ctx)
+	if !ok {
+		// default to apiserver audiences
+		requestedAudiences = a.implicitAuds
+	}
+
+	auds := kauthenticator.Audiences(tokenAudiences).Intersect(requestedAudiences)
+	if len(auds) == 0 && len(a.implicitAuds) != 0 {
+		return nil, false, fmt.Errorf("token audiences %q is invalid for the target audiences %q", tokenAudiences, requestedAudiences)
```

Comment:
I'll allow this because I think it will never happen, but the oidc example suggests `return nil, false, nil`

Reply:
It's copied one-to-one from the service account authenticator.